src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs

```diff
@@ -305,6 +320,151 @@ private string GetField(FieldExpression f)
         {
             return _query.Metadata.GetIndexFieldName(f, _query.QueryParameters).Value;
         }
+        
+        private IQueryMatch OrderBy(IQueryMatch match, List<(QueryExpression Expression, OrderByFieldType FieldType, bool Ascending)> orders, int take)
+        {
+            switch (orders.Count)
+            {
+                //Note: we want use generics up to 3 comparers. This way we gonna avoid virtual calls in most cases.
+                case 1:
+                {
+                    if (orders[0].Expression is not FieldExpression fe) throw new Exception();
+                    var id = GetFieldIdInIndex(GetField(fe));
+                    var orderTypeField = OrderTypeFieldConverter(orders[0].FieldType);
+
+                    match = orders[0].Ascending
+                        ? _searcher.OrderByAscending(match, id, orderTypeField, take)
+                        : _searcher.OrderByDescending(match, id, orderTypeField, take);
+
+
+                    return match;
+                }
+                case 2:
+                {
+                    var firstOrder = orders[0];
+                    var secondOrder = orders[1];
+                    if (firstOrder.Expression is not FieldExpression first || secondOrder.Expression is not FieldExpression second)
+                    {
+                        throw new Exception();
```

Comment:
Same here. BTW Shouldn't we move the exceptions throwing to dedicated methods? I believe that this method is gonna be called quite frequently. @redknightlois?